pyfarm/jobtypes/core/jobtype.py

```diff
@@ -890,6 +890,96 @@ def set_task_progress(self, task, progress):
                     "%r." % (task["id"], error))
                 raise
 
+    @inlineCallbacks
+    def set_task_started_now(self, task):
```

Comment:
Hmm, is there any reason we don't want to call this `set_task_time_started` so you could specify a specific time (including 'now').  I know the usual behavior should be to set `time_started` to now but it may also be useful to set specific times as well.